src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/ZipVerifiers.java

```diff
@@ -0,0 +1,125 @@
+package uk.gov.hmcts.reform.bulkscanprocessor.tasks.processor;
+
+import java.io.IOException;
+import java.security.InvalidKeyException;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.PublicKey;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+import static com.google.common.io.ByteStreams.toByteArray;
+
+
+public class ZipVerifiers {
+
+    public static final String DOCUMENTS_ZIP = "documents.zip";
+    public static final String SIGNATURE_SIG = "signature.sig";
+
+    private ZipVerifiers() {
+    }
+
+    public static Function<ZipStreamWithSignature, ZipInputStream> getPreprocessor(
+        String signatureAlgorithm
+    ) {
+        if ("sha256withrsa".equalsIgnoreCase(signatureAlgorithm)) {
```

Comment:
I think this implementation is not correct. Either you will have digital sig verification enabled or disabled. You won't be supporting multiple algorithms.

Reply:
There is no extra code in implementing it this way and it lets you change algorithm in future if needed. I do not see why it's not correct. Which part of the spec does it violate?

Reply:
whole thing is around enabling/disabling and not around algorithm choice. and definitely you should not code for future requirements as it might change.

Reply:
This allows enabling/disbaling + flexibility at no extra cost. You have a different opinion which I simply disagree with ;-)